lang/backend/src/result.rs

```diff
@@ -5,4 +5,8 @@ use thiserror::Error;
 pub enum BackendError {
     #[error("Impossible: {0}")]
     Impossible(String),
+    #[error("Code generation error: {0}")]
+    CodegenError(String),
 }
+
+pub type BackendResult<T> = Result<T, BackendError>;
```

Comment:
This type should use `T = ()`, and you don't currently use this type synonym in the implementation.